SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestSoftAssertHandler.cpp

```diff
@@ -0,0 +1,393 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialFunctionalTestSoftAssertHandler.h"
+#include "Logging/LogMacros.h"
+#include "Misc/AssertionMacros.h"
+#include "SpatialFunctionalTest.h"
+#include "SpatialFunctionalTestFlowController.h"
+#include "SpatialGDKFunctionalTestsPrivate.h"
+#include "VisualLogger/VisualLogger.h"
+
+namespace
+{
+template <typename T>
+bool Compare(const T& A, EComparisonMethod Operator, const T& B)
+{
+	bool bPassed = false;
+	switch (Operator)
+	{
+	case EComparisonMethod::Equal_To:
+		bPassed = A == B;
+		break;
+	case EComparisonMethod::Not_Equal_To:
+		bPassed = A != B;
+		break;
+	case EComparisonMethod::Greater_Than_Or_Equal_To:
+		bPassed = A >= B;
+		break;
+	case EComparisonMethod::Less_Than_Or_Equal_To:
+		bPassed = A <= B;
+		break;
+	case EComparisonMethod::Greater_Than:
+		bPassed = A > B;
+		break;
+	case EComparisonMethod::Less_Than:
+		bPassed = A < B;
+		break;
```

Comment:
similar to below maybe a default `checkNoEntry()` ?

Reply:
I wasn't aware of this macro, I can add it